### PR TITLE
Create Poetry publish workflow for dev snapshots

### LINKED ISSUE #112: Poetry publish should support dev version
- Right now we have workflows that allow to release the repo and publish to PyPI.
-  We would like to have something in between (Probably expanding the publish action/workflow) that allows to publish as a SNAPSHOT/dev version.

### PR DESCRIPTION
**Pull request by disrupted**
Closes #112

### CODE REVIEW

File: .github/workflows/README.md [hunk 1]
```diff
@@ -643,7 +644,57 @@ jobs:
...
+
+### Prerequisites
+
+Your Python project needs to be set up with Poetry and contain a `pyproject.toml` file to use this workflow.
+
+### Dependencies
+
+This workflow is built from multiple composite actions listed below:
+
+- [python-poetry-publish-pypi](https://github.com/bakdata/ci-templates/tree/main/actions/python-poetry-bump-version)
```
Comment (irux) [inline .github/workflows/README.md:662]: Don't forget to change the name here :)

File: .github/workflows/README.md [hunk 2]
```diff
@@ -627,7 +628,7 @@ These secrets define the pypi token that allow the GitHub action to release the
 ### Calling the workflow
 
 ```yaml
-name: Call this reusable workflow
+name: Publish
```
Comment (irux) [inline .github/workflows/README.md:631]: Maybe something more specific like Publish snapshot ?

**Comment by irux** [comment]
Did you test it ?

File: .github/workflows/README.md (hunk 1 above)
Comment (disrupted) [reply .github/workflows/README.md:662]: d'oh, thanks!

File: .github/workflows/README.md (hunk 2 above)
Comment (disrupted) [reply .github/workflows/README.md:631]: this is referring to the regular publish workflow, I named the new one as "Publish snapshot" already (see below)

**Comment by disrupted** [comment]
> Did you test it ?

not sure anymore, I've tested so many CI PRs recently, will do so in the playground now

**Review by irux — APPROVED** [review]
LGTM

**Pull request merged**